pyproject.toml

```diff
@@ -1,3 +1,142 @@
+]
+dependencies = [
+    "httpx ~= 0.27",
+]
+
+[project.urls]
+"Hompage" = "https://python-telegram-bot.org"
+"Documentation" = "https://docs.python-telegram-bot.org"
+"Bug Tracker" = "https://github.com/python-telegram-bot/python-telegram-bot/issues"
+"Source Code" = "https://github.com/python-telegram-bot/python-telegram-bot"
+"News" = "https://t.me/pythontelegrambotchannel"
+"Changelog" = "https://docs.python-telegram-bot.org/en/stable/changelog.html"
+
+[project.optional-dependencies]
+# Make sure to install those as additional_dependencies in the
+# pre-commit hooks for pylint & mypy
+# Also update the readme accordingly
+#
+# When dependencies release new versions and tests succeed, we should try to expand the allowed
+# versions and only increase the lower bound if necessary
+#
+# When adding new groups, make sure to update `ext` and `all` accordingly
+
+# Optional dependencies for production
+all = [
+    "python-telegram-bot[ext,http2,passport,socks]",
+]
+callback-data = [
+    # Cachetools doesn't have a strict stability policy. Let's be cautious for now.
+    "cachetools~=5.3.3",
+]
+ext = [
+    "python-telegram-bot[callback-data,job-queue,rate-limiter,webhooks]",
+]
+http2 = [
+    "httpx[http2]",
+]
+job-queue = [
+    # APS doesn't have a strict stability policy. Let's be cautious for now.
+    "APScheduler~=3.10.4",
+    # pytz is required by APS and just needs the lower bound due to #2120
+    "pytz>=2018.6",
+]
+passport = [
+    "cryptography!=3.4,!=3.4.1,!=3.4.2,!=3.4.3,>=39.0.1",
+]
+rate-limiter = [
+    "aiolimiter~=1.1.0",
+]
+socks = [
+    "httpx[socks]",
+]
+webhooks = [
+    # tornado is rather stable, but let's not allow the next mayor release without prior testing
+    "tornado~=6.4",
+]
+
+# Development only
+dev-all = [
+    "python-telegram-bot[all,dev,docs]",
+]
+dev = [
+    # needed for pre-commit hooks in the git commit command
+    "pre-commit",
+    # required for building the wheels for releases
+    "build",
+    # For the test suite
+    "pytest==8.2.1",
+    # needed because pytest doesn't come with native support for coroutines as tests
+    "pytest-asyncio==0.21.2",
+    # xdist runs tests in parallel
+    "pytest-xdist==3.6.1",
+    # Used for flaky tests (flaky decorator)
+    "flaky",
+    # used in test_official for parsing tg docs
+    "beautifulsoup4",
+]
+docs = [
+    "sphinx==7.3.7",
+    "furo==2024.5.6",
+    "furo-sphinx-search @ git+https://github.com/harshil21/furo-sphinx-search@v0.2.0.1",
+    "sphinx-paramlinks==0.6.0",
+    "sphinxcontrib-mermaid==0.9.2",
+    "sphinx-copybutton==0.5.2",
+    "sphinx-inline-tabs==2023.4.21",
+]
+
+
+# HATCH
+[tool.hatch.version]
+# dynamically evaluatios the `__version__` variable in that file
+source = "code"
+path = "telegram/_version.py"
+search-paths = ["telegram"]
+
+[tool.hatch.build]
+packages = ["telegram"]
+
+[tool.hatch.metadata]
+# For the furo-sphinx-search dependency
+allow-direct-references = true
```

Comment:
what does this do? Allow installation from git+ urls?

Reply:
Exactly:)